tests/binary_tests/puppet.test.js

```diff
@@ -156,3 +160,24 @@ test('Run a real invest model', async () => {
     expect(await findByText(doc, 'Open Workspace'));
   });
 });
+
+// Test for duplicate application launch.
+// We have the binary path, so now let's launch a new subprocess with the same binary
+// The test is that the subprocess exits within a certain reasonable timeout.
+// Also verify that window 1 has focus.
+test('App re-launch will exit and focus on first instance', async () => {
+  await waitFor(() => {
+    expect(browser.isConnected()).toBeTruthy();
+  });
+
+  // Open another instance of the Workbench application.
+  // This should return quickly.  The test timeout is there in case the new i
+  // process hangs for some reason.
+  const otherElectronProcess = spawnSync(
+    `"${binaryPath}"`, [`--remote-debugging-port=${PORT}`],
+    { shell: true }
+  );
+
+  // When another instance is already open, we expect an exit code of 1.
+  expect(otherElectronProcess.status).toBe(1);
```

Comment:
I know the puppeteer test is not running on Mac on GHA (though it will again soon, @emlys figured that out in #116 ), but I'm curious if you tried this on your Mac and what happens? Does this second `spawnSync` trigger the 'second-instance' event? And if so, what actually happens to `otherElectronProcess`? Does it also exit with code 1?

Reply:
`spawnSync` starts the process and then waits for it to exit, returning a process object that we can then inspect (which we use for the exit code status assertion).  So by the time we reach this assertion, `otherElectronProcess` should absolutely have exited, and I'd expect the first electron process to still be hanging around until jest kills it at the end of the test.

On my mac, this executes as expected (and does appear reliable), but the other test hasn't been running reliably yet ... haven't figured out why.

Reply:
Oh right, `spawnSync` is sync, so the fact that it gets to the assertion with an exit code is probably evidence that the 'second-instance' event did it's thing.